src/subwindows/singleCameraSettingsDialog.cpp

```diff
@@ -583,63 +705,63 @@ SingleCameraSettingsDialog::~SingleCameraSettingsDialog() = default;
 
 
 void SingleCameraSettingsDialog::onSetImageROIwidth(int val) {
-    singleCamera->setImageROIwidth(val);
+//    imageROIwidthInputBox->editingFinished();
+    camera->setImageROIwidth(val);
     // NOTE: qt will not consider the programmatic change of the value as a user event to handle
     updateImageROISettingsMax();
     updateImageROISettingsValues();
     updateCamImageRegionsWidget();
 }
 
 void SingleCameraSettingsDialog::onSetImageROIheight(int val) {
-    singleCamera->setImageROIheight(val);
+    camera->setImageROIheight(val);
     updateImageROISettingsMax();
     updateImageROISettingsValues();
     updateCamImageRegionsWidget();
 }
 
 void SingleCameraSettingsDialog::onSetImageROIoffsetX(int val) {
-    singleCamera->setImageROIoffsetX(val);
+    camera->setImageROIoffsetX(val);
     updateImageROISettingsMax();
     updateImageROISettingsValues();
     updateCamImageRegionsWidget();
 }
 
 void SingleCameraSettingsDialog::onSetImageROIoffsetY(int val) {
-    singleCamera->setImageROIoffsetY(val);
+    camera->setImageROIoffsetY(val);
     updateImageROISettingsMax();
     updateImageROISettingsValues();
     updateCamImageRegionsWidget();
 }
 
-void SingleCameraSettingsDialog::updateImageROISettingsMin(int binningVal) {
-    imageROIwidthInputBox->setMinimum(minImageSize/binningVal);
-    imageROIwidthInputBox->setSingleStep(imageSizeChangeSingleStep/binningVal);
-    imageROIheightInputBox->setMinimum(minImageSize/binningVal);
-    imageROIheightInputBox->setSingleStep(imageSizeChangeSingleStep/binningVal);
-    imageROIoffsetXInputBox->setMinimum(0);
-    imageROIoffsetXInputBox->setSingleStep(imageSizeChangeSingleStep/binningVal);
-    imageROIoffsetYInputBox->setMinimum(0);
-    imageROIoffsetYInputBox->setSingleStep(imageSizeChangeSingleStep/binningVal);
-}
-
 void SingleCameraSettingsDialog::updateImageROISettingsMax() {
-    imageROIwidthInputBox->setMaximum(singleCamera->getImageROIwidthMax());
-    imageROIheightInputBox->setMaximum(singleCamera->getImageROIheightMax());
-    imageROIoffsetXInputBox->setMaximum(singleCamera->getImageROIwidthMax() -singleCamera->getImageROIwidth());
-    imageROIoffsetYInputBox->setMaximum(singleCamera->getImageROIheightMax() -singleCamera->getImageROIheight());
-
-    imageROIwidthMaxLabel->setText(QString("/ ") + QString::number(singleCamera->getImageROIwidthMax()));
-    imageROIheightMaxLabel->setText(QString("/ ") + QString::number(singleCamera->getImageROIheightMax()));
-    imageROIoffsetXMaxLabel->setText(QString("/ ") + QString::number(singleCamera->getImageROIwidthMax() -singleCamera->getImageROIwidth()));
-    imageROIoffsetYMaxLabel->setText(QString("/ ") + QString::number(singleCamera->getImageROIheightMax() -singleCamera->getImageROIheight()));
+    if(!camera->isOpen()) {
+        imageROIwidthInputBox->setMaximum(std::numeric_limits<short>::max());
```

Comment:
Why are these needed if the camera is not open?

Reply:
Agree, I removed it from the single camera version, but note that once we migrate single and stereo camera settings windows, this must be taken care of (the if(!camera->isOpen()... parts will be needed), as there the cameras can be opened only after the settings dialog is drawn, in a different step